### PR TITLE
[tests] add tests to edge handler template

### PR DESCRIPTION
**Pull request by EndangeredMassa**
The Edge Handler wrapping logic we add during `vc dev` was not testable. This PR refactors it to be testable and adds some basic tests.

### CODE REVIEW

**Review by TooTallNate — DISMISSED** [review]

File: packages/node/src/edge-functions/edge-handler.ts [hunk 1]
```diff
@@ -69,13 +69,28 @@ async function compileUserCode(
 
       // user code
       ${compiledFile.text};
+      const userEdgeHandler = module.exports.default;
```
Comment (cb1kenobi) [inline packages/node/src/edge-functions/edge-handler.ts:72]: **question (non-blocking):** I follow that the ref to `module.exports.default` was moved from the template into here, but I don't get why it needed to be moved here. In theory, `registerFetchListener()` would be eval'd in the same context and thus resolve the same `module.exports.default`.

**Review by cb1kenobi — DISMISSED** [review]
LGTM, tests ran locally.

File: packages/node/src/edge-functions/edge-handler.ts (hunk 1 above)
Comment (EndangeredMassa) [reply packages/node/src/edge-functions/edge-handler.ts:72]: The `module.exports.default` was looking for the user's code from `compiledFile`. I put it right after the `compiledFile` interpolation to make that relationship visual.

The `edgeHandlerTemplate` is not related to this `moduled.exports.default`. The template comes later.

**Review by cb1kenobi — APPROVED** [review]

**Review by TooTallNate — APPROVED** [review]
